geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala

```diff
@@ -104,45 +103,60 @@ object AccumuloDataStoreFactory extends GeoMesaDataStoreInfo {
 
   def buildAccumuloConnector(params: java.util.Map[String, _ <: Serializable]): Connector = {
     if (MockParam.lookup(params)) {
       throw new IllegalArgumentException("Mock Accumulo connections are not supported")
     }
 
-    val instance = InstanceIdParam.lookup(params)
-    val zookeepers = ZookeepersParam.lookup(params)
-    // NB: For those wanting to set this via JAVA_OPTS, this key is "instance.zookeeper.timeout" in Accumulo 1.6.x.
-    val timeout = GeoMesaSystemProperties.getProperty(ClientProperty.INSTANCE_ZK_TIMEOUT.getKey)
+    def lookup[T <: AnyRef](param: GeoMesaParam[T], fallback: => Option[T]): T =
+      param.lookupOpt(params).orElse(fallback).getOrElse {
+        throw new IOException(s"Parameter ${param.key} is required: ${param.description}")
+      }
+
+    lazy val config = AccumuloClientConfig.load()
+    val instance = lookup(InstanceIdParam, config.instance)
+    val zookeepers = lookup(ZookeepersParam, config.zookeepers)
+
     val conf = ClientConfiguration.create().withInstance(instance).withZkHosts(zookeepers)
-    if (timeout != null) {
+    ZookeeperTimeoutParam.lookupOpt(params).orElse(config.zkTimeout).foreach { timeout =>
       conf.`with`(ClientProperty.INSTANCE_ZK_TIMEOUT, timeout)
     }
 
-    val user = UserParam.lookup(params)
-    val password = PasswordParam.lookup(params)
-    val keytabPath = KeytabPathParam.lookup(params)
+    val user = lookup(UserParam, config.principal)
+
+    if (PasswordParam.exists(params) && KeytabPathParam.exists(params)) {
+      throw new IllegalArgumentException(
+        s"'${PasswordParam.key}' and '${KeytabPathParam.key}' are mutually exclusive, but are both set")
+    }
+
+    val authType =
+      if (PasswordParam.exists(params)) {
+        AccumuloClientConfig.PasswordAuthType
+      } else if (KeytabPathParam.exists(params)) {
+        AccumuloClientConfig.KerberosAuthType
+      } else {
+        config.authType.map(_.trim.toLowerCase(Locale.US)).getOrElse {
+          throw new IOException(s"Parameter ${PasswordParam.key} is required: ${PasswordParam.description}")
+        }
+      }
 
     // build authentication token according to how we are authenticating
-    val auth: AuthenticationToken = if (password != null && keytabPath == null) {
-      new PasswordToken(password.getBytes(StandardCharsets.UTF_8))
-    } else if (password == null && keytabPath != null) {
+    val auth: AuthenticationToken = if (authType == AccumuloClientConfig.PasswordAuthType) {
+      new PasswordToken(lookup(PasswordParam, config.token).getBytes(StandardCharsets.UTF_8))
+    } else if (authType == AccumuloClientConfig.KerberosAuthType) {
       // explicitly enable SASL for kerberos connections
       // this shouldn't be required if Accumulo client.conf is set appropriately, but it doesn't seem to work
       conf.withSasl(true)
-      val file = new java.io.File(keytabPath)
+      val file = new java.io.File(lookup(KeytabPathParam, config.token))
```

Comment:
(Random question) Have we seen Accumulo 2.0 in a Kerberized environment?  

If not, is it worth noting that in the documentation somehow?

Reply:
we have not... we have some caveats on how it was tested already here: https://www.geomesa.org/documentation/user/accumulo/kerberos.html#development-testing